File: src/bin/lc1.cpp
#include <assert.h>
#include <map>
#include <vector>
using std::map;
using std::vector;
vector<int> twoSum(vector<int> &nums, int target);
int main(int argc, char const *argv[]) {
    /* code */
    auto v = vector<int>{3, 3};
    auto target = 6;
    auto rst = twoSum(v, target);
    assert(rst[0] == 0);
    assert(rst[1] == 1);
    return 0;
}

vector<int> twoSum(vector<int> &nums, int target) {
    auto rst = vector<int>{};
    map<unsigned int, int> hm;
    for (auto i = 0; i < nums.size(); i++) {
        int left = target - nums[i];
        auto fd = hm.find(left);
        if (fd != hm.end()) {
            rst.push_back(hm[left]);
            rst.push_back(i);
            break;
        } else {
            hm[nums[i]] = i;
        }
    }
    return rst;
}

File: src/bin/lc1.rs

fn main(){
    let nums=vec![3,3];
    let target=6;
    println!("{:?}",two_sum(nums, target));
}

fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    use std::collections::HashMap;
    let mut rst=vec![];
    let mut hs:HashMap<i32,usize>=HashMap::new();
    for   (idx,num) in nums.iter().enumerate(){
        if let Some(a)=hs.get(&(target-*num)){
                rst.push(*a as i32);
                rst.push(idx as i32);
                break;
        } else{
            hs.insert(*num, idx);
        }
    }
    rst
       
}